packages/google_maps_flutter/google_maps_flutter_web/lib/src/google_maps_controller.dart

```diff
@@ -413,6 +436,113 @@ class GoogleMapController {
     return _markersController?.isInfoWindowShown(markerId) ?? false;
   }
 
+  // Add My Location widget to right bottom
+  HtmlElement _createMyLocationButton() {
+    final HtmlElement controlDiv = DivElement();
+    controlDiv.style.marginRight = '10px';
+
+    final HtmlElement firstChild = ButtonElement();
+    firstChild.className = 'gm-control-active';
+    firstChild.style.backgroundColor = '#fff';
+    firstChild.style.border = 'none';
+    firstChild.style.outline = 'none';
+    firstChild.style.width = '40px';
+    firstChild.style.height = '40px';
+    firstChild.style.borderRadius = '2px';
+    firstChild.style.boxShadow = '0 1px 4px rgba(0,0,0,0.3)';
+    firstChild.style.cursor = 'pointer';
+    firstChild.style.padding = '8px';
+    controlDiv.append(firstChild);
+
+    final HtmlElement secondChild = DivElement();
+    secondChild.style.width = '24px';
+    secondChild.style.height = '24px';
+    secondChild.style.backgroundImage =
+        'url(https://maps.gstatic.com/tactile/mylocation/mylocation-sprite-2x.png)';
+    secondChild.style.backgroundSize = '240px 24px';
+    secondChild.style.backgroundPosition = '0px 0px';
+    secondChild.style.backgroundRepeat = 'no-repeat';
+    secondChild.id = 'you_location_img';
+    firstChild.append(secondChild);
+
+    firstChild.addEventListener('click', (_) {
+      String imgX = '0';
+      // Add animation when find current location
+      final Timer timer =
+          Timer.periodic(const Duration(milliseconds: 500), (_) {
+        imgX = (imgX == '-24') ? '0' : '-24';
+        document.getElementById('you_location_img')?.style.backgroundPosition =
+            '${imgX}px 0px';
+      });
+      // Find and move to current location
+      _moveToCurrentLocation().then((_) {
+        timer.cancel();
+        document.getElementById('you_location_img')?.style.backgroundPosition =
+            '-192px 0px';
+      });
+    });
+
+    return controlDiv;
+  }
+
+  // Get current location
+  Future<LatLng> _getCurrentLocation() async {
+    final Geoposition location =
+        await window.navigator.geolocation.getCurrentPosition();
```

Comment:
This method will not update the user's location if they move around, unless something else on the map changes. I think this should use the `watchPosition` API, that returns a `Stream<Geoposition>` and then respond to the events from that stream. Some changes to this PR:

* If the user wants to render the MyLocation dot: subscribe to the `watchPosition` Stream, and on each `Geolocation` event update the Marker.
  * The `first` event of the `watchPosition` Stream can be used to remove the "waiting" animation class from the button, for example, and to get ready the marker that needs to be rendered.
* If the user wants to stop rendering their location, it's easy to remove whatever subscription to the Stream that we do, and whatever cleanup is needed.

Some documentation:

* See: https://api.dart.dev/stable/2.18.6/dart-html/Geolocation/watchPosition.html
* Async in Dart - Streams: https://dart.dev/tutorials/language/streams